### ISSUE TITLE
Taskomatic failing after update from 2022.06

### ISSUE THREAD

**Issue by SSchott** [uyuni-project/uyuni#6088]
I just did an update and the web platform is not coming up. The Taskomatic service is not starting, and I get in the log ` /var/log/rhn/rhn_taskomatic_daemon.log` the following:
```
2022-11-02 11:30:18,765 [Thread-42] FATAL com.redhat.rhn.taskomatic.core.TaskomaticDaemon - javax/persistence/PersistenceException
```

Any ideas?

**Comment by SSchott** [comment]
Noticed that `zypper up` prints:
```
The following 28 package updates will NOT be installed:
  cobbler dom4j golang-github-QubitProducts-exporter_exporter hibernate-types netty objectweb-asm pgjdbc-ng prometheus-postgres_exporter spacewalk-backend
  spacewalk-backend-app spacewalk-backend-applet spacewalk-backend-config-files spacewalk-backend-config-files-common spacewalk-backend-config-files-tool
  spacewalk-backend-iss spacewalk-backend-iss-export spacewalk-backend-package-push-server spacewalk-backend-server spacewalk-backend-sql spacewalk-backend-sql-postgresql
  spacewalk-backend-tools spacewalk-backend-xml-export-libs spacewalk-backend-xmlrpc spacewalk-java spacewalk-java-lib spacewalk-setup spacewalk-taskomatic
  supportutils-plugin-salt
```

if I try to force updates on java, it offers:

```
zypper up spacewalk-java
Loading repository data...
Reading installed packages...
Resolving package dependencies...

Problem: the to be installed spacewalk-java-4.4.1-220400.1.15.uyuni2.noarch requires 'glassfish-activation-api', but this requirement cannot be provided
  not installable providers: glassfish-activation-api-1.2.0-1.6.uyuni2.noarch[uyuni-server-stable]
 Solution 1: Following actions will be done:
  install glassfish-activation-api-1.2.0-1.6.uyuni2.noarch from vendor obs://build.opensuse.org/systemsmanagement:Uyuni
    replacing gnu-jaf-1.1.1-2.41.noarch from vendor SUSE LLC <https://www.suse.com/>
  install objectweb-asm-9.3-1.13.uyuni2.noarch from vendor obs://build.opensuse.org/systemsmanagement:Uyuni
    replacing objectweb-asm-7.2-1.63.noarch from vendor SUSE LLC <https://www.suse.com/>
  install dom4j-1.6.1-11.1.uyuni2.noarch from vendor obs://build.opensuse.org/systemsmanagement:Uyuni
    replacing dom4j-1.6.1-10.12.noarch from vendor SUSE LLC <https://www.suse.com/>
 Solution 2: do not install spacewalk-java-4.4.1-220400.1.15.uyuni2.noarch
 Solution 3: break spacewalk-java-4.4.1-220400.1.15.uyuni2.noarch by ignoring some of its dependencies
```

Dunno if I should follow one of these options.

**Comment by marmau06** [comment]
Hello
same situation for me...
Installed 2022.10 from 2022.06
Installed patches (https://www.uyuni-project.org/pages/patches.html)

Rebooted server
taskomatic not running, same messages
Web interface unavailable...

:(

**Comment by SSchott** [comment]
Managed to...fix (?) taskomatic by installing the optional spacewalk-java and additional packages. That broke cobbler, which after fiddeling, got fixed when running manually with the migration option. Now all modules are loading, but, the web UI still is not there, with tomcat showing: 
```
02-Nov-2022 15:50:41.139 SEVERE [main] org.apache.catalina.startup.HostConfig.deployDirectory Error deploying web application directory [/srv/tomcat/webapps/rhn]
        java.lang.IllegalStateException: Error starting child
                at org.apache.catalina.core.ContainerBase.addChildInternal(ContainerBase.java:720)
                at org.apache.catalina.core.ContainerBase.addChild(ContainerBase.java:690)
                at org.apache.catalina.core.StandardHost.addChild(StandardHost.java:705)
                at org.apache.catalina.startup.HostConfig.deployDirectory(HostConfig.java:1132)
                at org.apache.catalina.startup.HostConfig$DeployDirectory.run(HostConfig.java:1865)
                at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
                at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
                at org.apache.tomcat.util.threads.InlineExecutorService.execute(InlineExecutorService.java:75)
                at java.base/java.util.concurrent.AbstractExecutorService.submit(AbstractExecutorService.java:118)
                at org.apache.catalina.startup.HostConfig.deployDirectories(HostConfig.java:1044)
                at org.apache.catalina.startup.HostConfig.deployApps(HostConfig.java:429)
                at org.apache.catalina.startup.HostConfig.start(HostConfig.java:1575)
                at org.apache.catalina.startup.HostConfig.lifecycleEvent(HostConfig.java:309)
                at org.apache.catalina.util.LifecycleBase.fireLifecycleEvent(LifecycleBase.java:123)
                at org.apache.catalina.util.LifecycleBase.setStateInternal(LifecycleBase.java:423)
                at org.apache.catalina.util.LifecycleBase.setState(LifecycleBase.java:366)
                at org.apache.catalina.core.ContainerBase.startInternal(ContainerBase.java:936)
                at org.apache.catalina.core.StandardHost.startInternal(StandardHost.java:841)
                at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:183)
                at org.apache.catalina.core.ContainerBase$StartChild.call(ContainerBase.java:1384)
                at org.apache.catalina.core.ContainerBase$StartChild.call(ContainerBase.java:1374)
                at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
                at org.apache.tomcat.util.threads.InlineExecutorService.execute(InlineExecutorService.java:75)
                at java.base/java.util.concurrent.AbstractExecutorService.submit(AbstractExecutorService.java:140)
                at org.apache.catalina.core.ContainerBase.startInternal(ContainerBase.java:909)
                at org.apache.catalina.core.StandardEngine.startInternal(StandardEngine.java:262)
                at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:183)
                at org.apache.catalina.core.StandardService.startInternal(StandardService.java:421)
                at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:183)
                at org.apache.catalina.core.StandardServer.startInternal(StandardServer.java:930)
                at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:183)
                at org.apache.catalina.startup.Catalina.start(Catalina.java:633)
                at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
                at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
                at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
                at java.base/java.lang.reflect.Method.invoke(Method.java:566)
                at org.apache.catalina.startup.Bootstrap.start(Bootstrap.java:343)
                at org.apache.catalina.startup.Bootstrap.main(Bootstrap.java:474)
        Caused by: org.apache.catalina.LifecycleException: Failed to start component [StandardEngine[Catalina].StandardHost[localhost].StandardContext[/rhn]]
                at org.apache.catalina.util.LifecycleBase.handleSubClassException(LifecycleBase.java:440)
                at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:198)
                at org.apache.catalina.core.ContainerBase.addChildInternal(ContainerBase.java:717)
                ... 37 more
        Caused by: java.lang.NullPointerException
                at org.apache.tomcat.util.scan.StandardJarScanner.process(StandardJarScanner.java:382)
                at org.apache.tomcat.util.scan.StandardJarScanner.scan(StandardJarScanner.java:195)
                at org.apache.catalina.startup.ContextConfig.processJarsForWebFragments(ContextConfig.java:1971)
                at org.apache.catalina.startup.ContextConfig.webConfig(ContextConfig.java:1129)
                at org.apache.catalina.startup.ContextConfig.configureStart(ContextConfig.java:775)
                at org.apache.catalina.startup.ContextConfig.lifecycleEvent(ContextConfig.java:301)
                at org.apache.catalina.util.LifecycleBase.fireLifecycleEvent(LifecycleBase.java:123)
                at org.apache.catalina.core.StandardContext.startInternal(StandardContext.java:5044)
                at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:183)
                ... 38 more

```

**Comment by marmau06** [comment]
Hello,
This morning, I just managed the update.. and taskomatic fails...
Obviously, I have new RHEL9 servers in my scope, so I need this upgrade !!! :(

**Comment by SSchott** [comment]
Which Java version do you have?

**Comment by SSchott** [comment]
Fixed Tomcat finally, taking ideas from #5278 . Sooooo, for the record, and as like the 3 times before Uyuni broke, I am not 100% sure why it broke, but starting from 2022.06, this (I think! trying to filter from a sea of commands) did it for me:
```
spacewalk-service stop
#Add repo to pass patch to fix repo update in 2022.06
zypper addrepo https://download.opensuse.org/repositories/systemsmanagement:/Uyuni:/Stable:/Patches/openSUSE_Leap_15.4/systemsmanagement:Uyuni:Stable:Patches.repo
zypper refresh
zypper up
#Fix Taskomatic
zypper in -f spacewalk-taskomatic
zypper up spacewalk-java
#Fix Cobbler
/usr/bin/cobblerd --enable-automigration
#Fixes Tomcat broken links
zypper install netty-4.1.75-150200.4.9.1.x86_64
zypper up pgjdbc-ng
spacewalk-service start
```

**Comment by santeri3700** [comment]
Did you update Zypper before updating Uyuni and allow vendor change(s) like instructed in the update notes? https://www.uyuni-project.org/doc/2022.10/release-notes-uyuni-server.html#_update_notes

**Comment by marmau06** [comment]
Hello, as usual on my side, the issue has been located between the chair and the keyboard...
THanks to vmware, snapshot allowed me to restart from beginning..
So, as mentionned in both documentation:
- I updated zypper
- I udated package using "--allow-vendor-change" found in release notes.
- I restarted the server, Uyuni was Ok
- I installed patches
- I restarted the server, Uyuni is now Ok.
So I think that I just missed the "--allow-vendor-change" switch...

Sorry to disturb you..
If I can bring other elements to help you, just ask...

Have a nice day.

**Comment by juliogonzalez** [comment]
> Fixed Tomcat finally, taking ideas from #5278 . Sooooo, for the record, and as like the 3 times before Uyuni broke, I am not 100% sure why it broke, but starting from 2022.06, this (I think! trying to filter from a sea of commands) did it for me:
> 
> ```
> spacewalk-service stop
> #Add repo to pass patch to fix repo update in 2022.06
> zypper addrepo https://download.opensuse.org/repositories/systemsmanagement:/Uyuni:/Stable:/Patches/openSUSE_Leap_15.4/systemsmanagement:Uyuni:Stable:Patches.repo
> zypper refresh
> zypper up
> #Fix Taskomatic
> zypper in -f spacewalk-taskomatic
> zypper up spacewalk-java
> #Fix Cobbler
> /usr/bin/cobblerd --enable-automigration
> #Fixes Tomcat broken links
> zypper install netty-4.1.75-150200.4.9.1.x86_64
> zypper up pgjdbc-ng
> spacewalk-service start
> ```

Cobbler had problems migrating files provided by old cobbler versions. So if you installed Uyuni a while ago, that could explain it.

There are patches now to handle that, release that Friday.

There's a procedure for the patches, do not call `zypper in` for individual packages. Apply the update to 2022.10 as stated on the release notes (as mentioned, there's a special step this time), then apply the patches as explained at https://www.uyuni-project.org/pages/patches.html

**Comment by SSchott** [comment]
Thanks for the clarification. Hope the info helps others with problems in the topic. I'll close it for now, as my system is working.
